includes/reader-activation/class-reader-data.php

```diff
@@ -348,17 +373,14 @@ public static function check_newsletter_subscription( $timestamp, $data ) {
 		if ( ! empty( $is_newsletter_subscriber ) && gettype( $is_newsletter_subscriber ) === 'string' ) {
 			$is_newsletter_subscriber = json_decode( $is_newsletter_subscriber );
 		}
-		// Bail if reader is already a newsletter subscriber.
-		if ( $is_newsletter_subscriber ) {
-			return;
-		}
 		// Check if the user is subscribed to a newsletter.
 		if ( ! class_exists( '\Newspack_Newsletters' ) || ! class_exists( '\Newspack_Newsletters_Subscription' ) ) {
 			return;
 		}
 		$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $data['email'] );
 		if ( ! is_wp_error( $subscribed_lists ) && ! empty( $subscribed_lists ) && is_array( $subscribed_lists ) ) {
```

Comment:
Shouldn't there be an else here, where `is_newsletter_subscriber` would be set to  `false`? After I've unsubscribed from all lists, this value (as seen in `localStorage`) was not updated: `newsletter_subscribed_lists` was `[]`, but `is_newsletter_subscriber` was `true`.